src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/SqlKernelsExtensionTests.cs

```diff
@@ -90,6 +90,17 @@ private static IEnumerable<IEnumerable<IEnumerable<(string name, object value)>>
             } while (reader.NextResult());
         }
 
+        [Fact]
+        public void can_alias_duplicate_column_names_in_sql_results()
+        {
+            var testNames = new string[] { "apple", "banana", "apple", "orange", "orange", "BANANA", "orange", "BaNaNa" };
+            var expectedAliases = new string[] { "apple", "banana", "apple (1)", "orange", "orange (1)", "BANANA", "orange (2)", "BaNaNa" };
```

Comment:
It seems like we should do one of the following:

* Also number the first occurrence of "apple", or
* Start the numbering at 2.

Reply:
By testing this method in isolation, we're not actually guaranteeing that the end-to-end works. A more robust approach would be to verify against the actual serialized output.

Reply:
I'll do the second option. Doing the first would require two passes through the names array.

Reply:
If one approach is clearer or more common than the other, I wouldn't sweat an extra loop through the array. The perf hit there is negligible especially on top of network I/O and serialization costs.